packages/asset-service/src/generateAssetData/ethereum/idleVaults.ts

```diff
@@ -18,23 +19,37 @@ const explorerData = {
 }
 
 const getIdleVaults = async (): Promise<Asset[]> => {
+  const alchemyApiKey = process.env.ALCHEMY_API_KEY
   const vaults: IdleVault[] = await idleSdk.getVaults()
 
-  return vaults.map((vault: IdleVault) => {
+  const parsedVaults = []
+  for (const vault of vaults) {
+    const alchemyUrl = `https://eth-mainnet.g.alchemy.com/v2/${alchemyApiKey}`
+    const alchemyPayload = JSON.stringify({
+      jsonrpc: '2.0',
+      method: 'alchemy_getTokenMetadata',
+      params: [vault.address],
+      id: 42,
```

Comment:
https://docs.alchemy.com/docs/how-to-get-token-metadata#step-2-write-the-token-metadata-querying-script-1